backends/s3/s3_test.go

```diff
@@ -51,6 +52,9 @@ func getBackend(ctx context.Context, t *testing.T) (b *Backend) {
 	require.NoError(t, err)
 
 	cleanup := func() {
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
My thought here was that the outer-scope `ctx` might have been cancelled by the time `cleanup` is called. Should I try and use the outer-scope one regardless? Or make my intent clearer with something like `if ctx.Err() != nil { ctx = ... }`?

Reply:
That outer scope will indeed likely be cancelled by the time this runs, because of the defer. The t.Cleanup that was called earlier would have been run after this cleanup function, so things would probably have worked as expected. But since you changed the t.Cleanup into a defer, the context will be closed before this is run.